read.c

```diff
@@ -536,7 +540,7 @@ static int processAggregateItem(redisReader *r) {
             if (r->fn && r->fn->createArray)
                 obj = r->fn->createArray(cur,elements);
             else
-                obj = (void*)(long)cur->type;
+                obj = (void*)(intptr_t)cur->type;
```

Comment:
Must use an integral type of the same size as pointer to avoid the (sensible) warning here.